[docs] Explain the `shouldDisableTime` migration more in-depth (#8348)
mui/mui-x: docs/data/migration/migration-pickers-v5/migration-pickers-v5.md

At some point, the mobile pickers should have a prop allowing to have an editable field without opening the modal.
:::

### ✅ Rename or refactor `shouldDisableTime` prop

The `shouldDisableTime` prop signature has been changed.
Previously it did receive `value` as a `number` of hours, minutes, or seconds. Now it will receive the date object (based on the used adapter).

> This will allow more powerful usage and will enable the introduction of different time selection views.

Author: What do you think @flaviendelangle 
```suggestion
This will allow more powerful usage and introducing of a digital time selection view.
```

Reviewer: How about

```suggestion
This will allow more powerful usage and will be compatible with the future digital time selection view.
```

> It was done to allow for more powerful usage as well as enable the introduction of different time selection views.

Reviewer: ```suggestion
This was done to allow for more powerful usage as well as enable the introduction of different time selection views.
```

```suggestion
This will allow more powerful usage and will enable the introduction of different time selection views.
```
